src/DotNet.Testcontainers/Containers/Modules/TestcontainersContainer.cs

```diff
@@ -218,6 +214,7 @@ public virtual async ValueTask DisposeAsync()
       await cleanOrStopTask.ConfigureAwait(false);
 
       this.semaphoreSlim.Dispose();
+      this.client.Dispose();
```

Comment:
I have added the semaphore again, but kept disposing the client here. To me it looks like this was missing before as we are initializing the client in the constructor (line 29), but are not disposing it in the `Dispose` method.